### PR TITLE
HIVE-24792: Potential thread leak in Operation

### PR DESCRIPTION
**Pull request by dengzhhu653**

### CODE REVIEW

**Comment by dengzhhu653** [comment]
Move the operation log cleanup logic into a standalone method...
@prasanthj  could you please take a look? thank you

File: service/src/java/org/apache/hive/service/cli/operation/Operation.java [hunk 1]
```diff
@@ -326,6 +326,7 @@ protected synchronized void cleanupOperationLog(final long operationLogCleanupDe
         if (operationLogCleanupDelayMs > 0) {
           scheduledExecutorService.schedule(new OperationLogCleaner(operationLog), operationLogCleanupDelayMs,
             TimeUnit.MILLISECONDS);
+          scheduledExecutorService.shutdown();
```
Comment (pvary) [inline service/src/java/org/apache/hive/service/cli/operation/Operation.java:329]: If I understand correctly, then the pool is created regardless of the config. We might want to close outside of the conditional expressions. What do you think?

Comment (pvary) [reply service/src/java/org/apache/hive/service/cli/operation/Operation.java:329]: Or even beter, move the executor creation to here too....

Comment (dengzhhu653) [reply service/src/java/org/apache/hive/service/cli/operation/Operation.java:329]: Yes, thanks very much for the advice.

**Review by pvary — APPROVED** [review]
+1 pending tests

**Pull request merged**